### ISSUE TITLE
arangodb hitting image sz limit ?

### ISSUE THREAD

**Issue by eyberg** [nanovms/nanos#514]
just noticed it's a +600mb binary

idk if this is a mkfs || something in ops that needs to change - just putting it here for now

```
eyberg@dungeon:~/ar2$ ops run -p 8529 -c config.json arangod
Finding dependent shared libs
[--configuration arangod.conf]
2019/02/15 15:20:41
*exec.ExitError exit status 1
/home/tijoytom/go/src/github.com/nanovms/ops/lepton/image.go:29 (0x76ca0f)
/home/tijoytom/go/src/github.com/nanovms/ops/ops.go:143 (0x7daba8)
/home/tijoytom/go/src/github.com/nanovms/ops/ops.go:113 (0x7d96b4)
/home/tijoytom/go/src/github.com/spf13/cobra/command.go:766 (0x7cfe1c)
/home/tijoytom/go/src/github.com/spf13/cobra/command.go:852 (0x7d09bd)
/home/tijoytom/go/src/github.com/spf13/cobra/command.go:800 (0x7d069b)
/home/tijoytom/go/src/github.com/nanovms/ops/ops.go:522 (0x7df507)
/usr/local/go/src/runtime/proc.go:201 (0x42de97)
        main: fn()
/usr/local/go/src/runtime/asm_amd64.s:1333 (0x459051)
        goexit: BYTE    $0x90   // NOP

panic: exit status 1

goroutine 1 [running]:
main.panicOnError(0x9461a0, 0xc0002fe050)
        /home/tijoytom/go/src/github.com/nanovms/ops/ops.go:46 +0xde
main.buildImages(0xc0001509a0)
        /home/tijoytom/go/src/github.com/nanovms/ops/ops.go:144 +0xd6
main.runCommandHandler(0xc00011b180, 0xc0001143c0, 0x1, 0x5)
        /home/tijoytom/go/src/github.com/nanovms/ops/ops.go:113 +0x5e4
github.com/spf13/cobra.(*Command).execute(0xc00011b180, 0xc000114320, 0x5, 0x5, 0xc00011b180, 0xc000114320)
        /home/tijoytom/go/src/github.com/spf13/cobra/command.go:766 +0x2cc
github.com/spf13/cobra.(*Command).ExecuteC(0xc00018ac80, 0x8, 0x0, 0xc00018a780)
        /home/tijoytom/go/src/github.com/spf13/cobra/command.go:852 +0x2fd
github.com/spf13/cobra.(*Command).Execute(0xc00018ac80, 0xc000133ee8, 0x1)
        /home/tijoytom/go/src/github.com/spf13/cobra/command.go:800 +0x2b
main.main()
        /home/tijoytom/go/src/github.com/nanovms/ops/ops.go:522 +0x1257
eyberg@dungeon:~/ar2$ ls -lh
total 628M
-rwxrwxr-x 1 eyberg eyberg 628M Feb 15 15:17 arangod
-rw-r--r-- 1 eyberg eyberg 1.7K Feb 15 15:18 arangod.conf
-rw-rw-r-- 1 eyberg eyberg  104 Feb 15 15:20 config.json
drwxrwxr-x 8 eyberg eyberg 4.0K Feb 15 15:19 js
drwxrwxr-x 2 eyberg eyberg 4.0K Feb 15 15:19 tmp
eyberg@dungeon:~/ar2$ cat config.json
{
  "Args": ["--configuration", "arangod.conf"],
  "Files": ["arangod.conf"],
  "Dirs": ["tmp", "js"]
}
```

this is a bit weird cause i was curious and saw that mysqld was 725mb and it successfully builds the image so idk if that's the case or not

```
eyberg@dungeon:~/mys$ ls -lh mysqld
-rwxrwxr-x 1 eyberg eyberg 725M Nov 26 15:18 mysqld
eyberg@dungeon:~/mys$ ls -lh image
-rw-rw-r-- 1 eyberg eyberg 736M Feb 15 15:26 image
```

**Comment by eyberg** [comment]
this pages faults after looking at overcommit && then trying to mmap an anonymous page ->

```
1 readlink
1 readlink: "/etc/malloc.conf"
1 direct return: -22, rsp 0x76e77cc8
1 open
1 open: "/proc/sys/vm/overcommit_memory", flags 80000, mode 0
1 "/proc/sys/vm/overcommit_memory" - not found
1 direct return: -2, rsp 0x76e78d28
1 mmap
1 mmap: target 0x0000000000000000, size 0x200000, len 0x200000, prot 0x3, flags 0x22, fd -1, offset 0x0
1    anon target: 0x8bb600000, len: 0x200000 (given size: 0x200000)
1 direct return: 37503369216, rsp 0x76e78b68

Unhandled page fault in kernel mode: cpu: 0
 interrupt: 000000000000000e (Page fault)
     frame: 0000000100e05400
error code: 0000000000000002
   address: fffffffffffffff8

   rax: 00000008bb600000
   rbx: 0000000000000003
   rcx: 0000000000000012
   rdx: ffffffffffffffff
   rsi: 0000000000200000
   rdi: 0000000000000000
   rbp: 0000000000002000
   rsp: 0000000076e78d20
    r8: 7000000000000000
    r9: 00000000003fffff
   r10: 0000000000000015
   r11: 0000000000000246
   r12: 00000008bb600000
   r13: 000000000000001c
   r14: 0000000000200000
   r15: 0000000500229170
   rip: 00000000022fb425
rflags: 0000000000000212
    ss: 0000000000000023
    cs: 000000000000002b
    ds: 0000000000000000
    es: 0000000000000000
fsbase: 000000088f2ea240
gsbase: 0000000000000000

frame trace:

stack trace:
010100000000004e
```

**Comment by francescolavra** [comment]
The above page fault is not reproducible with current master. Using ArangoDB 3.7.6 and newer, the application hits a division-by-zero fault, which is fixed by #1419.
```
francesco@ubuntu:~/.ops/instances/arangodb$ cat config.json 
{
  "Files": ["arangod.conf"],
  "Dirs": ["/usr/share/arangodb3"],
  "BaseVolumeSz": "200m",
  "RunConfig": {
    "Ports": ["8529"]
  },
  "Args": ["arangod", "--configuration", "arangod.conf", "--log.output", "+"]
}
```
```
francesco@ubuntu:~/.ops/instances/arangodb$ cat arangod.conf 
# ArangoDB configuration file
# 
# Documentation: 
# https://www.arangodb.com/docs/stable/administration-configuration.html
# 

[database]
directory = /var/lib/arangodb3

[server]
# Specify the endpoint for HTTP requests by clients.
#  tcp://ipv4-address:port
#  tcp://[ipv6-address]:port
#  ssl://ipv4-address:port
#  ssl://[ipv6-address]:port
#  unix:///path/to/socket
#
# Examples:
#   endpoint = tcp://0.0.0.0:8529
#   endpoint = tcp://127.0.0.1:8529
#   endpoint = tcp://localhost:8529
#   endpoint = tcp://myserver.arangodb.com:8529
#   endpoint = tcp://[::]:8529
#   endpoint = tcp://[fe80::21a:5df1:aede:98cf]:8529
#
endpoint = tcp://0.0.0.0:8529
storage-engine = auto

# reuse a port on restart or wait until it is freed by the operating system
# reuse-address = false

authentication = true

# number of maximal server threads. use 0 to make arangod determine the
# number of threads automatically, based on available CPUs
# maximal-threads = 0

# gather server statistics
statistics = true

# the user and group are normally set in the start script
# uid = arangodb
# gid = arangodb

[javascript]
startup-directory = /usr/share/arangodb3/js
app-path = /var/lib/arangodb3-apps
# app-path = //arangodb3/apps

# number of V8 contexts available for JavaScript execution. use 0 to 
# make arangod determine the number of contexts automatically.
# v8-contexts = 0

[foxx]
# enable Foxx queues in the server
# queues = true

# interval (seconds) to use for polling jobs in Foxx queues
# queues-poll-interval = 1

[log]
level = info
file = arangod.log

[cluster]

[rocksdb]
# encryption-keyfile=/your-encryption-file
```
```
francesco@ubuntu:~/.ops/instances/arangodb$ ops run arangod -c config.json
booting /home/francesco/.ops/images/arangod.img ...
en1: assigned 10.0.2.15
2021-03-02T12:15:03Z [2] INFO [e52b0] ArangoDB 3.7.8 [linux] 64bit, using jemalloc, build tags/v3.7.8-0-gefd7b1049d, VPack 0.1.33, RocksDB 6.8.0, ICU 64.2, V8 7.9.317, OpenSSL 1.1.1i  8 Dec 2020
2021-03-02T12:15:03Z [2] INFO [25362] {memory} Available physical memory: 2132934656 bytes, available cores: 1
2021-03-02T12:15:03Z [2] INFO [0f48d] {communication} IPv6 support seems to be disabled
2021-03-02T12:15:03Z [2] INFO [43396] {authentication} Jwt secret not specified, generating...
2021-03-02T12:15:03Z [2] INFO [144fe] using storage engine 'rocksdb'
2021-03-02T12:15:03Z [2] INFO [3bb7d] {cluster} Starting up with role SINGLE
2021-03-02T12:15:03Z [2] INFO [a1c60] {syscall} file-descriptors (nofiles) hard limit is 65536, soft limit is 65536
2021-03-02T12:15:03Z [2] INFO [24783] created database directory '/var/lib/arangodb3'
2021-03-02T12:15:03Z [2] INFO [3844e] {authentication} Authentication is turned on (system only), authentication for unix sockets is turned on
2021-03-02T12:15:04Z [2] INFO [1bf74] created --javascript.app-path directory '/var/lib/arangodb3-apps'
2021-03-02T12:15:04Z [2] INFO [e6460] created base application directory '/var/lib/arangodb3-apps/_db'
2021-03-02T12:15:04Z [2] INFO [c1b63] {arangosearch} ArangoSearch maintenance: [1..1] commit thread(s), [1..1] consolidation thread(s)
2021-03-02T12:15:04Z [2] INFO [6ea38] using endpoint 'http+tcp://0.0.0.0:8529' for non-encrypted requests
2021-03-02T12:15:04Z [2] INFO [857d7] {authentication} Creating user "root"
2021-03-02T12:15:04Z [2] INFO [cf3f4] ArangoDB (version 3.7.8 [linux]) is ready for business. Have fun!
```
To connect to the DB server, the arangosh tool can be used.

**Comment by eyberg** [comment]
made a pkg - closing
